float64 x
float64 y
float64 z

---

float64 result_1
float64 result_2

